addon-test-support/ember-mocha/setup-test.js

```diff
@@ -30,37 +21,35 @@ function setupPauseTest(context) {
   };
 }
 
-export default function setupTest(options) {
+export default function setupTest(_options) {
   let originalContext;
   let beforeEachHooks = [];
   let afterEachHooks = [];
+  let options = _options === undefined ? { waitForSettled: true } : Object.assign({ waitForSettled: true }, _options);
 
   beforeEach(function() {
-    originalContext = _assign({}, this);
+    originalContext = Object.assign({}, this);
+    let context = new Proxy(this, {});
+    this._emberContext = context;
 
-    return setupContext(this, options)
+    return setupContext(this._emberContext, options)
       .then(() => setupPauseTest(this))
       .then(() => chainHooks(beforeEachHooks, this));
   });
 
   afterEach(function() {
     return chainHooks(afterEachHooks, this)
-      .then(() => teardownContext(this))
+      .then(() => teardownContext(this._emberContext, options))
       .then(() => {
         // delete any extraneous properties
         for (let key in this) {
           if (!(key in originalContext)) {
-            // starting from Ember 3.13 this seems to be necessary
-            if (teardownMandatorySetter) {
-              teardownMandatorySetter(this, key);
-            }
-
             delete this[key];
           }
         }
 
         // copy over the original values
-        _assign(this, originalContext);
+        Object.assign(this, originalContext);
```

Comment:
Are we explicitly dropping IE11 support in this release? IMHO, that should be decoupled from updating @ember/test-helpers.

Reply:
@rwjblue would the babel transpiler not handle this for us?

Reply:
> @rwjblue would the babel transpiler not handle this for us?

No, not by default. It is _possible_ to configure this to transpile into core-js poly fills, but that requires the application has the polyfill installed/configured (which we can't assume). Additionally, we can't force the app to add the polyfill only in tests because then tests would pass in browsers that would never actually work in production/deployment.

Reply:
So unfortunately with the `Proxy` solution in order to support the latest test-helpers we will either have to drop IE support or use https://github.com/GoogleChrome/proxy-polyfill. LMK if it still makes sense to decouple these.

Reply:
@rwjblue let me know what you would like to do regarding this? Should I use the proxy-polyfill or should we just drop IE support in this PR?